static/src/javascripts/projects/commercial/modules/dfp/prepare-googletag.js

```diff
@@ -15,17 +15,58 @@ import onSlotLoad from 'commercial/modules/dfp/on-slot-load';
 import { fillAdvertSlots } from 'commercial/modules/dfp/fill-advert-slots';
 import refreshOnResize from 'commercial/modules/dfp/refresh-on-resize';
 import { adFreeSlotRemove } from 'commercial/modules/close-disabled-slots';
+import { register } from 'commercial/modules/messenger';
 import {
     addTag,
     setListeners,
 } from 'commercial/modules/dfp/performance-logging';
-import 'commercial/modules/messenger/type';
-import 'commercial/modules/messenger/get-stylesheet';
-import 'commercial/modules/messenger/resize';
-import 'commercial/modules/messenger/scroll';
-import 'commercial/modules/messenger/viewport';
-import 'commercial/modules/messenger/click';
-import 'commercial/modules/messenger/background';
+import { setType } from 'commercial/modules/messenger/type';
+import { getStyles } from 'commercial/modules/messenger/get-stylesheet';
+import { resize } from 'commercial/modules/messenger/resize';
+import { onMessage as onMessageScroll } from 'commercial/modules/messenger/scroll';
+import { onMessage as onMessageViewport } from 'commercial/modules/messenger/viewport';
+import { sendClick } from 'commercial/modules/messenger/click';
+import { setBackground } from 'commercial/modules/messenger/background';
+
+register('type', (specs: ?string, ret, iframe) =>
+    setType(specs, iframe && iframe.closest('.js-ad-slot'))
+);
+
+register('get-styles', (specs): ?Array<any> => {
+    if (specs) {
+        return getStyles(specs, document.styleSheets);
+    }
+});
+
+register('resize', (specs, ret, iframe) => {
+    if (iframe && specs) {
+        const adSlot = iframe && iframe.closest('.js-ad-slot');
+        return resize(specs, iframe, adSlot);
+    }
+});
+
+register('scroll', onMessageScroll, {
+    persist: true,
+});
+
+register('viewport', onMessageViewport, {
+    persist: true,
+});
+
+register('click', (linkName, ret, iframe = {}) =>
+    sendClick(
+        iframe.closest('.js-ad-slot') || {
+            id: 'unknown',
+        },
+        linkName || ''
+    )
+);
+
+register('background', (specs, ret, iframe): ?Promise<any> => {
+    if (iframe && specs) {
+        return setBackground(specs, iframe.closest('.js-ad-slot'));
+    }
+});
```

Comment:
I'm not completely sure, so I put them like this to be on the safe side. I'll investigate.